neos/neos-development-collection
# !!! BUGFIX: Prevent busting the configuration cache version when a user logs in or is updated

**What I did**

Without this patch the configuration version was flushed and raised after every login or any other change to an account, always forcing the browsers to reload nodetype schemas and other configuration-requests which used the configuration version as cache-buster parameter. So if a user would reload the Neos backend after someone else logged in or logs in and out, the backend loading time would be much slower, as the nodetype schema and xliff data would have to be regenerated and downloaded skipping the possibly already cached version in the browser.

Also each role combination of users now has their own version identifier and therefore can share the resulting nodetype schema, xliff, etc.

Before everyone had the version of the user that last logged in or had any other change to their account as every change would have flushed the configuration version.

**How I did it**

* Removed the `AccountPostEventListener` which only flushed the `Neos_Neos_Configuration_Version`. This could be breaking if anyone relied on it.
* Make sure every role combination has their own configuration version.

**How to verify it**

* Log in with first user and note down configuration version query parameter of nodetype request
* Log in with second user with same roles and compare configuration version, should be the same
* Reload with first user and make sure the version is still the same
